pkg/neonvm/ipam/allocate.go

```diff
@@ -41,20 +57,17 @@ func doAcquire(
 }
 
 func doRelease(
-	_ context.Context,
 	ipRange RangeConfiguration,
 	reservation []whereaboutstypes.IPReservation,
-	vmName string,
-	vmNamespace string,
+	vmName types.NamespacedName,
 ) (net.IPNet, []whereaboutstypes.IPReservation, error) {
 	// reduce whereabouts logging
 	whereaboutslogging.SetLogLevel("error")
 
-	vmID := fmt.Sprintf("%s/%s", vmNamespace, vmName)
 	_, ipnet, _ := net.ParseCIDR(ipRange.Range)
 
 	// try to release IP for given VM
-	newReservation, ip, err := whereaboutsallocate.IterateForDeallocation(reservation, vmID, getMatchingIPReservationIndex)
+	newReservation, ip, err := whereaboutsallocate.IterateForDeallocation(reservation, vmName.String(), getMatchingIPReservationIndex)
```

Comment:
Just double-checking, this is idempotent, right? i.e., trying to release an IP that's already been released will not return error?

(thinking about failure modes for https://github.com/neondatabase/autoscaling/pull/1278/commits/04be1a67a66a72b6d21bca254e41f62bee8d4982)

Reply:
Yeah, added a test for this.

Reply:
But it was actually failing, fixed it now in the latest commit.